js/Glacier2/callback/MockAlarmClock.ts

```diff
@@ -0,0 +1,23 @@
+import { Ice } from "@zeroc/ice";
+import { EarlyRiser } from "./AlarmClock.js";
+
+export class MockAlarmClock extends EarlyRiser.AlarmClock {
+    private _needMoreTime = true;
+    private _resolver: (value: void | PromiseLike<void>) => void;
+
+    public StopPressed = new Promise<void>((resolve, reject) => (this._resolver = resolve));
```

Comment:
Is StopPressed created when MockAlarmClock is created or the first time StopPressed is used?

Reply:
When it is created.

The generated JavaScript looks like:

```
export class MockAlarmClock extends EarlyRiser.AlarmClock {
    constructor() {
        super(...arguments);
        this._needMoreTime = true;
        this.StopPressed = new Promise((resolve, reject) => (this._resolver = resolve));
    }
```